packages/core-mobile/app/new/routes/(signedIn)/(tabs)/browser/index.tsx

```diff
@@ -1,21 +1,143 @@
-import React from 'react'
-import { View, Text } from '@avalabs/k2-alpine'
-import BlurredBarsContentLayout from 'common/components/BlurredBarsContentLayout'
+import { ANIMATED, View } from '@avalabs/k2-alpine'
+import { useBrowserContext } from 'features/browser/BrowserContext'
+import { BrowserControls } from 'features/browser/components/BrowserControls'
+import { BrowserSnapshot } from 'features/browser/components/BrowserSnapshot'
+import {
+  BrowserTab,
+  BrowserTabRef
+} from 'features/browser/components/BrowserTab'
+import { Discover } from 'features/browser/components/Discover'
+import React, { useEffect, useMemo, useState } from 'react'
+import Animated, { useAnimatedStyle, withTiming } from 'react-native-reanimated'
+import { useSelector } from 'react-redux'
+import {
+  selectActiveTab,
+  selectAllTabs,
+  selectIsTabEmpty,
+  Tab
+} from 'store/browser'
+
+const Browser = (): React.ReactNode => {
+  const { browserRefs } = useBrowserContext()
+  const activeTab = useSelector(selectActiveTab)
+  const allTabs = useSelector(selectAllTabs)
+  const showEmptyTab = useSelector(selectIsTabEmpty)
+
+  const [tabs, setTabs] = useState<Tab[]>([])
+
+  useEffect(() => {
+    tabs.forEach(tab => {
+      if (browserRefs?.current && !browserRefs.current[tab.id]?.current) {
+        browserRefs.current[tab.id] = React.createRef<BrowserTabRef>()
+      }
+    })
+  }, [tabs, browserRefs])
```

Comment:
let's also clean up unused refs or else refs will keep building up as we close out tabs.

Reply:
yes but let's also remove unused refs or else the list of refs will keep building up